pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -438,12 +408,64 @@ private ParsedTimestamp parseBackendTimestamp(String str) throws SQLException {
       return LocalTime.parse(s);
     } catch (DateTimeParseException nfe) {
       throw new PSQLException(
-          GT.tr("Bad value for type timestamp/date/time: {1}", s),
+          GT.tr("Bad value for type timestamp/date/time: {0}", s),
           PSQLState.BAD_DATETIME_FORMAT, nfe);
     }
 
   }
 
+  /**
+   * Returns the offset time object matching the given bytes with Oid#TIMETZ or Oid#TIME.
+   *
+   * @param bytes The binary encoded TIMETZ/TIME value.
+   * @return The parsed offset time object.
+   * @throws PSQLException If binary format could not be parsed.
+   */
+  public OffsetTime toOffsetTimeBin(byte[] bytes) throws PSQLException {
+    if ((bytes.length != 8 && bytes.length != 12)) {
+      throw new PSQLException(GT.tr("Unsupported binary encoding of {0}.", "time"),
+          PSQLState.BAD_DATETIME_FORMAT);
+    }
+
+    // question: use default timezone with offset of current day here?
+    ZoneOffset timeOffset = ZoneOffset.UTC;
```

Comment:
The question here is: Should we use UTC or the local zone offset. The problem is without a date a timezone is useless, so I think UTC is correct here. Tis also matches OffsetDateTime behaviour

Reply:
I see how "just use UTC" might sound good, however, I wonder if `time -> OffsetTime` conversion should be allowed at all.

What if we just throw an exception like "please use `getObject(.., LocalTime.class)` instead"?

Reply:
This is exactly what I asked myself!
Currently it is possible also to get a `OffsetDateTime` also for `TIMESTAMP`, although this has the same (semantic) problem. Of course `TIMESTAMPTZ` in Postgres is always stored and returned in UTC (so actually both are the same on wire), so here it is fine to set always UTC as zone offset.
`TIMETZ` is special, because actually here Postgres saves the timezone. And also here we have the other assymmetry: You can do `getObject(LocalTime.class)` on `TIMETZ`. If we disallow getting `OffsetTime` from `TIME`, we must also disallow getting `LocalTime` from `TIMETZ`.

Actually at several places in code it mentions the JDBC 4.2 standard that seems to specify how the java.time classes are mapped to SQL types, so we should do what they say. I tried to download it, but it gave me just the API, but no spec about how types map. Do you have a copy of this PDF to check that is mentioned in several comments?

Reply:
> You can do getObject(LocalTime.class) on TIMETZ. If we disallow getting OffsetTime from TIME, we must also disallow getting LocalTime from TIMETZ.

In current code this fails in binary representation, but works in text transmission. A test is clearly missing here!

So in my opinion, we should explicitely forbid both. What do you think?